core/arch/arm/plat-ls/main.c

```diff
@@ -161,3 +162,34 @@ void main_secondary_init_gic(void)
 {
 	gic_cpu_init(&gic_data);
 }
+
+/*
+ * Issued when requesting to Secure Storage Key for secure storage.
+ *
+ * SiP Service Calls
+ *
+ * Register usage:
+ * r0/x0	SMC Function ID, OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY
+ */
+#define OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY			0xFF14
+#define OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY \
+	OPTEE_SMC_CALL_VAL(OPTEE_SMC_32, OPTEE_SMC_FAST_CALL, \
+			   OPTEE_SMC_OWNER_SIP, \
+			   OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY)
+
+void tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	uint8_t dummy[sizeof(hwkey->data)];
+
+	memset(dummy, 0x00, sizeof(hwkey->data));
+
+	get_hw_unique_key(OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY,
+			virt_to_phys(&hwkey->data[0]), sizeof(hwkey->data));
+
+	/* Invalidate cache to fetch data from external memory */
+	cache_op_inner(DCACHE_AREA_INVALIDATE,
+			&hwkey->data[0], sizeof(hwkey->data));
+
+	if (memcmp(hwkey->data, dummy, sizeof(hwkey->data)) != 0)
```

Comment:
Isn't there a better way to find out that the `OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY` call did not return a unique ID? (such as a "not implemented"/"failed" status?)

Reply:
I will work to apply your suggestion. till then updating the patch to if other comments are successfully disposed off.

Reply:
Accepted.
Made the changes to remove the dummy array.